From d906d854e233aead3ad7d1a9b578df39338cac22 Mon Sep 17 00:00:00 2001
From: Vincent Becker <vincentx.becker@intel.com>
Date: Fri, 28 Feb 2014 12:18:56 +0100
Subject: [PATCH 3/9] [REVERT ME][AUDIO] Set isVirtualInputDevice method as
 protected

BZ: 217971

Orig-Bug: 179042

isVirtualInputDevice is needed by the derived audio policy,
to check if the current input device is a virtual device.
This methods is currently private to the base policy class,
and can not be called from the daugther policy class.

Set protected access to the isVirtualInputDevice method, so
that it can be called from the derived policy.

Change-Id: I623d4b88ce0de13092b949eff7814d161ecd85eb
Category: device enablement
Domain: AUDIO_MEDIA-WIDI
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Vincent Becker <vincentx.becker@intel.com>
---
 include/hardware_legacy/AudioPolicyManagerBase.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/hardware_legacy/AudioPolicyManagerBase.h b/include/hardware_legacy/AudioPolicyManagerBase.h
index c5aac7b..eadd2b6 100644
--- a/include/hardware_legacy/AudioPolicyManagerBase.h
+++ b/include/hardware_legacy/AudioPolicyManagerBase.h
@@ -587,13 +587,14 @@ protected:
         uint32_t        mTestLatencyMs;
 #endif //AUDIO_POLICY_TEST
 
+        static bool isVirtualInputDevice(audio_devices_t device);
+
 private:
         static float volIndexToAmpl(audio_devices_t device, const StreamDescriptor& streamDesc,
                 int indexInUi);
         // updates device caching and output for streams that can influence the
         //    routing of notifications
         void handleNotificationRoutingForStream(AudioSystem::stream_type stream);
-        static bool isVirtualInputDevice(audio_devices_t device);
 };
 
 };
-- 
1.9.2

